src/Streamly/Internal/Data/Stream/IsStream.hs

```diff
@@ -599,55 +599,42 @@ module Streamly.Internal.Data.Stream.IsStream
-import Control.Monad.Trans.Class (MonadTrans(..))
-import Control.Monad.Trans.Control (MonadBaseControl)
-import Data.Either (isLeft, isRight)
-import Data.Functor.Identity (Identity (..))
 import Data.Kind (Type)
 import Data.Heap (Entry(..))
-import Data.Maybe (isJust, fromJust, isNothing)
-import Data.Void (Void)
+import Data.Maybe (isNothing)
 import Foreign.Storable (Storable)
-import Streamly.Internal.BaseCompat (fromLeft, fromRight)
 import Streamly.Internal.Data.Stream.Enumeration
        (Enumerable(..), enumerate, enumerateTo)
-import Streamly.Internal.Data.Fold.Types (Fold (..), Fold2 (..))
-import Streamly.Internal.Data.Parser (Parser (..))
-import Streamly.Internal.Data.Unfold.Types (Unfold)
-import Streamly.Internal.Data.Array.Storable.Foreign.Types (Array, writeNUnsafe)
-import Streamly.Internal.Data.SVar (MonadAsync, defState, Rate (..))
-import Streamly.Internal.Data.Stream.Ahead (AheadT, Ahead, ahead, aheadly)
+import Streamly.Internal.Data.Fold.Types (Fold (..))
+import Streamly.Internal.Data.Array.Storable.Foreign.Types (Array)
+import Streamly.Internal.Data.SVar (MonadAsync, Rate (..))
+import Streamly.Internal.Data.Stream.Ahead (AheadT, Ahead, aheadly)
 import Streamly.Internal.Data.Stream.Async
-       ( AsyncT, Async, WAsyncT, WAsync, mkAsync, async, asyncly, wAsync
+       ( AsyncT, Async, WAsyncT, WAsync, mkAsync, asyncly
        , wAsyncly)
 import Streamly.Internal.Data.Stream.Combinators
-      ( inspectMode, maxBuffer, maxThreads, maxYields, rate, avgRate, minRate
+      ( inspectMode, maxBuffer, maxThreads, rate, avgRate, minRate
       , maxRate, constRate)
 import Streamly.Internal.Data.Stream.Parallel
-       ( ParallelT, Parallel, parallel, parallely)
-import Streamly.Internal.Data.Stream.Prelude
-       (fromStreamS, toStreamS, concatFoldableWith, concatMapFoldableWith
-       , concatForFoldableWith)
-import Streamly.Internal.Data.Stream.StreamD (fromStreamD, toStreamD)
+       ( ParallelT, Parallel, parallely)
 import Streamly.Internal.Data.Stream.StreamK (IsStream((|:), consM), adapt)
 import Streamly.Internal.Data.Stream.Serial
-       ( SerialT, WSerialT, Serial, WSerial, serial, wSerial, serially
+       ( SerialT, WSerialT, Serial, WSerial, serially
        , wSerially)
 import Streamly.Internal.Data.Stream.Zip
        ( ZipSerialM, ZipSerial, ZipAsyncM, ZipAsync, zipSerially, zipAsyncly)
-import Streamly.Internal.Data.Pipe.Types (Pipe (..))
 import Streamly.Internal.Data.Time.Units
        ( AbsTime, MilliSecond64(..), addToAbsTime, toRelTime
-       , toAbsTime, TimeUnit64, RelTime64, addToAbsTime64)
-import Streamly.Internal.Data.IORef.Prim (Prim, IORef)
+       , toAbsTime, RelTime64)
 import Streamly.Internal.Data.Tuple.Strict (Tuple'(..))
 
+import Streamly.Internal.Data.Stream.IsStream.Generate
+import Streamly.Internal.Data.Stream.IsStream.Eliminate
+import Streamly.Internal.Data.Stream.IsStream.Transform
+import Streamly.Internal.Data.Stream.IsStream.Exceptions
```

Comment:
Maybe you could move these after the qualified imports. Not a big deal, you can ignore this.